src/process_transaction.c

```diff
@@ -27,7 +27,9 @@ void to_uppercase(char *str, unsigned char size) {
 void set_ledger_currency_name(char *currency) {
     for (size_t i = 0; i < sizeof(currencies_aliases) / sizeof(currencies_aliases[0]); i++) {
         if (!strcmp(currency, (char *) (PIC(currencies_aliases[i].foreign_name)))) {
-            strcpy(currency, (char *) (PIC(currencies_aliases[i].ledger_name)));
+            strlcpy(currency,
+                    (char *) (PIC(currencies_aliases[i].ledger_name)),
+                    strlen((char *) PIC(currencies_aliases[i].ledger_name)));
```

Comment:
this field should be the max size currency can be written into before overflowing

Reply:
Actually if think it should be something like `MIN(sizeof(<array>currency), strlen((char *) PIC(currencies_aliases[i].ledger_name)))`.
This size is [purely local](https://github.com/LedgerHQ/app-exchange/blob/develop/src/proto/protocol.pb.h#L40-L41) for now, I'll make it more public.